File: be/src/main/java/com/quangnt0000/be_modul/modal/DataLake/ReportTemplate.java
package com.quangnt0000.be_modul.modal.DataLake;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
@EntityListeners(AuditingEntityListener.class)

public class ReportTemplate {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    private String name;
    private String description;
    private String fileType;
    private String reportType;
    private String fileKey;
    private String reportId;

    @ManyToOne
    @JoinColumn(name = "employeeId")
    private Employee employee;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;

    @Builder.Default
    private Boolean deleted = false;

    @ManyToOne
    @JoinColumn(name = "reportCategoryId")
    private ReportCategory reportCategory;

}

File: fe-modul-report/src/features/report/types/reportTemplate.ts
export interface ReportTemplateRequest {
    id?: string;
    name?: string;
    description?: string;
    reportType?: string;
    file?: File;      
    reportId?: string;
    reportCategoryId?: string;
  }

  export interface ReportTemplateResponse {
    id?: string;
    name?: string;
    description?: string;
    fileType?: string;
    reportType?: string;
    fileKey?: string;
    reportId?: string;
    reportCategoryName?: string;
    employeeName?: string;
    createdAt: string;
  }
  